src/common/protocol.ts

```diff
@@ -150,6 +150,54 @@ export class Protocol {
 		}
 	}
 
+	toString(): string {
```

Comment:
let's update the tests to also check the `type` of each protocol, and add some unit tests for this method

Reply:
`type` checking is there. The only catch here is we don't have code path for `git://` at this moment.